encodings/runend/src/array.rs

```diff
@@ -248,10 +255,105 @@ impl StatisticsVTable<RunEndArray> for RunEndEncoding {
     }
 }
 
+impl RunEndArray {
+    fn true_count(&self) -> VortexResult<u64> {
+        let ends = self.ends().into_primitive()?;
+        let bools = self.values().into_bool()?.boolean_buffer();
+
+        Ok(match self.values().logical_validity() {
+            LogicalValidity::AllValid(_) => {
+                match_each_unsigned_integer_ptype!(ends.ptype(), |$P| {
+                    let mut begin = self.offset() as $P;
+                    ends
+                        .as_slice::<$P>()
+                        .iter()
+                        .zip_eq(bools.into_iter())
+                        .map(|(end, bool_value)| {
+                            let len = *end - begin;
+                            begin = *end;
+                            (len as u64) * (bool_value as u64)
+                        })
+                        .sum()
+                })
+            }
+            LogicalValidity::AllInvalid(_) => 0,
+            LogicalValidity::Array(is_valid) => {
+                let is_valid = is_valid.into_bool()?.boolean_buffer();
+                let mut is_valid = is_valid.set_indices();
+                match is_valid.next() {
+                    None => self.len() as u64,
+                    Some(valid_index) => {
+                        let mut true_count: u64 = 0;
+                        match_each_unsigned_integer_ptype!(ends.ptype(), |$P| {
+                            let offsetted_begin = <$P>::try_from(self.offset())?;
+                            let offsetted_len = <$P>::try_from(self.len() + self.offset())?;
```

Comment:
you don't need to adjust the length, all that has to happen is that last end has to be clamped at array.len().

Reply:
I think it is correct as is because the ends are stored as `true_end + offset`. I'm working directly on the unmodified ends; therefore, the length of the last interval is `min(ends[n-1], length + offset) - ends[n-2]`.

Reply:
I have gone in circles on this. The trimmed_ends_iterator does what you're doing here